src/Task/Exec.php

```diff
@@ -96,15 +102,26 @@ protected function ensureCommandRunsViaBash($command) {
     return '/bin/bash -c ' . escapeshellarg($command);
   }
 
-  /**
-   * {@inheritdoc}
-   */
-  protected function execute($process, $output_callback = NULL) {
-
-    // If PHAPP_ENV etc. is already set, be sure to keep that.
-    $process->inheritEnvironmentVariables(TRUE);
-
-    return parent::execute($process, $output_callback);
+    /**
+     * {@inheritdoc}
+     */
+  public function run() {
+      $env = [];
+      if ($phapp_env = getenv('PHAPP_ENV')) {
+          $env['PHAPP_ENV'] = $phapp_env;
```

Comment:
should not be needed. let's remove special-casing of PHAPP_ENV here.

Reply:
This is also needed because otherwise PHAPP_ENV is not in the env

Reply:
@fago I'm not exactly sure how inheritEnvironmentVariables() method used to work, but what we're actually doing is moving php env variables to bash env and it Process doesn't do that automatically.

Reply:
I see, we were making sure at least PHAPP_ENV  is set in bash environment. I think we can easily repalce this by simply having
> PHAPP_ENV={{ $ENV }}
as "environment" default value of phapp manifest. So when the loader is not customized, at least the variable is set automatically.